src/types/ErrorRate/types.ts

```diff
@@ -0,0 +1,38 @@
+import { ToleranceConfig } from '../ServerConfig';
+import { ThresholdStatus } from '../Health';
+
+/*
+Error Ratio for:
+  - Global: Inbound and Outbound requests
+  - Inbound Requests
+  - Outbound Requests
+ */
+export interface ErrorRatio {
+  global: { status: ThresholdStatus; protocol?: string; toleranceConfig?: ToleranceConfig };
+  inbound: { status: ThresholdStatus; protocol?: string; toleranceConfig?: ToleranceConfig };
+  outbound: { status: ThresholdStatus; protocol?: string; toleranceConfig?: ToleranceConfig };
+}
+
+/*
+Rate Interface:
+- The number of requests in t seconds requested by the user
+- The number of requests with error code
+- The ratio % of errors
+*/
+
+export interface Rate {
+  requestRate: number;
+  errorRate: number;
+  errorRatio: number;
+}
+
+/*
+RequestTolerance interface
+- Tolerance configuration applied
+- Requests error rate calculation for the tolerance Configuration
+
+*/
+export interface RequestTolerance {
+  tolerance: ToleranceConfig;
+  requests: { [key: string]: Rate | number };
```

Comment:
Good to add in the comments what is expected to be used as key here.
Perhaps for someone less familar is not obvious (i.e. I doubt if this is protocol/code).

```
requests: { [key: string]: Rate | number };
```